### PR TITLE
Unirepo

### LINKED ISSUE #162: master test runner
1. clone every submodule into a tmp directory
2. npm i
3. npm t
4. capture results so things blow up if any of the packages fail their tests

This will be kind of slow, but I think it makes sense to have this run from `npm test` every time. This will prevent subtle dependency update bugs that can happen when all the modules are working together.

### LINKED ISSUE #191: versioning for releases / development
thoughts on how to do this:

* do we want to pin dependencies for turf versions?
* how do we make sure that dependencies of modules don't require duplicate versions of the same module?

### LINKED ISSUE #232: sudo: false in all .travis.yml
Probably none of the turf repositories need to use sudo on Travis, so we should add `sudo: false` to all .travis.ymls for faster builds.

http://docs.travis-ci.com/user/workers/container-based-infrastructure/

### PR DESCRIPTION
**Pull request by tmcw**
* [x] Switch to a uni-repo method. All core modules will be in a single repo. Addons will be managed as such
* [x] Turf packages will still be bundled as split-out packages, like lodash modules or babel modules, and we'll take advantage of lerna to do that.
* [x] geometry-helpers these modules should be one module
* [x] `npm test` should run all package tests

Things this resolves

* Fixes #162 (makes tests easy to run)
* Fixes #232 (switches to circleci)
* Fixes #191 (lerna handles versioning)

Refs #306

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,6 +128,7 @@
     "eslint": "^0.15.0",
     "hcat": "0.0.5",
     "jsdoc": "^3.3.0-beta1",
+    "lerna": "https://github.com/kittens/lerna/archive/master.tar.gz",
```
Comment (morganherlocker) [inline package.json]: @tmcw Can we depend on lerna through npm? 

[babel reference](https://github.com/babel/babel/blob/master/package.json#L24)

Comment (tmcw) [inline package.json]: Yes, but the published version doesn't have `lerna run` yet

**Comment by tmcw** [comment]
Okay all you @morganherlocker to push the button

**Comment by migurski** [comment]
The internet salutes you.

**Comment by maxogden** [comment]
was there a discussion on this? just curious to learn more about what drove the decision to go unirepo

**Comment by tmcw** [comment]
Yep, discussion is at https://github.com/Turfjs/turf/issues/306 - coordinated multi-repo improvements like adding eslint and running tests were rough, and managing releases of all modules to coordinate with a release never found a solution - while the lerna/etc approach has been battle-tested

**Pull request merged**